game/plugin/navigation/door/src/door.plugin.kts

```diff
@@ -0,0 +1,212 @@
+import org.apollo.game.action.DistancedAction
+import org.apollo.game.model.entity.Player
+import org.apollo.game.model.entity.obj.DynamicGameObject
+import org.apollo.game.model.entity.EntityType
+import org.apollo.game.model.entity.obj.GameObject
+import org.apollo.game.model.Position
+import org.apollo.game.model.World
+import org.apollo.game.model.event.PlayerEvent
+import org.apollo.game.message.impl.ObjectActionMessage
+import org.apollo.net.message.Message
+
+enum class Orientation(val value: Int) {
+    WEST(0), NORTH(1), EAST(2), SOUTH(3);
+    companion object {
+        fun from(findValue: Int): Orientation = Orientation.values().first { it.value == findValue }
+    }
+}
+
+enum class DoorType {
+    LEFT, RIGHT, NOT_SUPPORTED
+}
+
+class Door(private val gameObject: GameObject) {
+
+    companion object {
+
+        /**
+         * Mapping to change orientation of a left hinged door
+         */
+        val LEFT_HINGE_ORIENTATION: HashMap<Orientation, Orientation> = hashMapOf(
+                Orientation.NORTH to Orientation.WEST,
+                Orientation.SOUTH to Orientation.EAST,
+                Orientation.WEST to Orientation.SOUTH,
+                Orientation.EAST to Orientation.NORTH
+        )
+
+        /**
+         * Mapping to change orientation of a right hinged door
+         */
+        val RIGHT_HINGE_ORIENTATION: HashMap<Orientation, Orientation> = hashMapOf(
+                Orientation.NORTH to Orientation.EAST,
+                Orientation.SOUTH to Orientation.WEST,
+                Orientation.WEST to Orientation.NORTH,
+                Orientation.EAST to Orientation.SOUTH
+        )
+
+        /**
+         * The list of doors that have already been toggled
+         */
+        val toggledDoors: HashMap<GameObject, GameObject> = hashMapOf()
+
+        /**
+         * The ids of all left hinged doors.
+         */
+        val LEFT_HINGED = setOf(1516, 1536, 1533)
+
+        /**
+         * The ids of all right hinged doors.
+         */
+        val RIGHT_HINGED = setOf(1519, 1530, 4465, 4467, 3014, 3017, 3018, 3019)
+
+        /**
+         * Find a given door in the world
+         * @param world The [World] the door lives in
+         * @param position The [Position] of the door
+         * @param objectId The [GameObject] id of the door
+         */
+        fun find(world: World, position: Position, objectId: Int): Door? {
+            val objects: Set<GameObject> = world.regionRepository.fromPosition(position).getEntities(position, EntityType.DYNAMIC_OBJECT, EntityType.STATIC_OBJECT)
+            val gameObject: GameObject? = objects.find { it.id == objectId }
```

Comment:
You can express this more concisely using the [elvis operator](https://kotlinlang.org/docs/reference/null-safety.html#elvis-operator):
```kotlin
val gameObject: GameObject = objects.find { it.id == objectId } ?: return null
return Door(gameObject)
```

Reply:
This can be reduced to

```kotlin
import org.apollo.game.plugins.api.*

val gameObject = region.findObject(position, id)
```

Since https://github.com/apollo-rsps/apollo/commit/cf0e3331e0fec57169f0fa0a6dd7be110e4e694f

Reply:
Nice!

Reply:
When storing region as a variable, are you suggesting
`Region region = world.regionRepository.fromPosition(position)`
As I am having issues defining this line. @garyttierney

Reply:
Yes,

```kotlin
val region = world.regionRepository.fromPosition(position)
val object = region.findObject(position, id)
```